Solution.vb

```diff
@@ -0,0 +1,47 @@
+﻿Imports System.IO
+Module Module1
+
+    Dim searchRoot As String
+    Dim ext As String
+    Dim subStr As String
+
+
+    Public Function GetAllSuitableFilePaths(searchRoot, ext, subStr) As String()
+
+        If searchRoot = "" Then
+            Console.WriteLine("Hodnota adresáře nemůže být prázdná")
+            Exit Function
+        ElseIf System.IO.Directory.Exists(searchRoot) = False Then
+            Console.WriteLine("Uvedený adresář neexistuje")
+            Exit Function
+        ElseIf ext = "" Then
+            Console.WriteLine("Hodnota přípony nemůže být prázdná")
+            Exit Function
+        End If
+
+        Dim Directory As New IO.DirectoryInfo(searchRoot)
+        Dim FileArray As IO.FileInfo() = Directory.GetFiles("*" & subStr & "*" & ext, SearchOption.AllDirectories)

```

Comment:
Tohle je supr řešení. Kdyby to bylo potřeba udělat bez této funkce a musel bys tu kontrolu a prohledání složek napsat sám ,věděl bys jak? Klidně napiš i druhou funkci, pokud bys to chtěl zkusit.